crates/bevy_math/src/cubic_splines.rs

```diff
@@ -472,22 +472,40 @@ impl<P: Point> CubicCurve<P> {
     /// A flexible iterator used to sample curves with arbitrary functions.
     ///
     /// This splits the curve into `subdivisions` of evenly spaced `t` values across the
-    /// length of the curve from start (t = 0) to end (t = 1), returning an iterator that evaluates
-    /// the curve with the supplied `sample_function` at each `t`.
+    /// length of the curve from start (t = 0) to end (t = n), where `n = self.segment_count()`,
+    /// returning an iterator evaluating the curve with the supplied `sample_function` at each `t`.
     ///
-    /// Given `subdivisions = 2`, this will split the curve into two lines, or three points, and
-    /// return an iterator over those three points, one at the start, middle, and end.
+    /// For `subdivisions = 2`, this will split the curve into two lines, or three points, and
+    /// return an iterator with 3 items, the three points, one at the start, middle, and end.
     #[inline]
-    pub fn iter_samples(
-        &self,
+    pub fn iter_samples<'a, 'b: 'a>(
+        &'b self,
         subdivisions: usize,
-        sample_function: fn(&Self, f32) -> P,
-    ) -> impl Iterator<Item = P> + '_ {
-        (0..=subdivisions).map(move |i| {
-            let segments = self.segments.len() as f32;
-            let t = i as f32 / subdivisions as f32 * segments;
-            sample_function(self, t)
-        })
+        mut sample_function: impl FnMut(&Self, f32) -> P + 'a,
+    ) -> impl Iterator<Item = P> + 'a {
+        self.iter_samples_t(subdivisions)
+            .map(move |t| sample_function(self, t))
+    }
+
+    /// Iterates over `subdivisions + 1` `t`s evenly spaced from the start of
+    /// the first curve segment to the end of the last segment.
+    #[inline]
+    fn iter_samples_t(&self, subdivisions: usize) -> impl ExactSizeIterator<Item = f32> {
```

Comment:
```suggestion
    fn iter_uniform(&self, subdivisions: usize) -> impl ExactSizeIterator<Item = f32> {
```

This is a bit of a nit, but I think this clarifies what this function does. It also fits with the nomenclature for splines. We currently only support uniform splines, but nonuniform splines sample the basis functions using a knot vector (instead of uniform spacing), and allow for more advanced shapes.

Reply:
I'd rather use the adverb here. I chose `iter_uniformly`. It's a bit odd, but given there is already a bunch of `iter_noun`, it's surprising to find a `iter_adjective`, I worry people would read "uniform" as a noun. With "uniformly" there is no ambiguity.